lib/led-matrix/src/LEDMatrix.cpp

```diff
@@ -0,0 +1,62 @@
+/* LED Matrix Source File */
+
+#include "LEDMatrix.h"
+
+// Constructor: An LED Matrix is defined by the pins it is connected to.
+LEDMatrix ::LEDMatrix(PinName R, PinName G, PinName B) : _RChannel(R), _GChannel(G), _BChannel(B) {}
+
+// Take in values from 0 to 255 for each color stream and maps them to a PWM signal.
+LEDMatrix ::void setColorRGB(uint8_t R, uint8_t G, uint8_t B) {
+  _RChannel.pulsewidth((1.0 / 255.0) * R);
+  _GChannel.pulsewidth((1.0 / 255.0) * G);
+  _BChannel.pulsewidth((1.0 / 255.0) * B);
+}
+
+// Just for convenience. Call setColorRGB().
+LEDMatrix ::void setColor(color c) {
+  switch (c) {
+    case color::RED:
+      setColorRGB(255, 0, 0);
+      break;
+    case color::GREEN:
+      setColorRGB(0, 255, 0);
+      break;
+    case color::BLUE:
+      setColorRGB(0, 0, 255);
+      break;
+  }
+}
+
+// Set the specified color, wait, turn off all LEDs, wait, and repeat until another function is called by the main.cpp.
+LEDMatrix ::void setFlashingColorRGB(uint8_t R, uint8_t G, uint8_t B, uint8_t frequency_hz) {
+  float flash_num  = frequency_hz * HW_BRIDGE::LEDMATRIX::NUM_FLASHES;  // Flash for 10sec
```

Comment:
I think flash duration should either be part of the API, or it should just flash indefinitely. I would say the latter is probably more appropriate so that ROS will just say FLASH and it'll keep flashing until it says STOP FLASHING

Reply:
I agree with melvin here even though i told u to do the opposite i think. So yeah, just make it flash . this requires adding creating a thread in here